addOns/commonlib/commonlib.gradle.kts

```diff
@@ -32,9 +37,40 @@ dependencies {
     api("com.fasterxml.jackson.datatype:jackson-datatype-jdk8")
     api("com.fasterxml.jackson.datatype:jackson-datatype-jsr310")
 
+    val antlrVersion = "4.13.0"
+    antlr("org.antlr:antlr4:$antlrVersion")
+    implementation("org.antlr:antlr4-runtime:$antlrVersion")
+
     implementation("commons-io:commons-io:2.16.1")
     implementation("org.apache.commons:commons-csv:1.10.0")
     implementation("org.apache.commons:commons-collections4:4.4")
 
     testImplementation(project(":testutils"))
 }
+
+val jsParserPkg = "org.zaproxy.addon.commonlib.parserapi.impl"
+val jsParserDir = jsParserPkg.replace('.', '/')
+val generateGrammarSource by tasks.existing(AntlrTask::class) {
+    val libDir = "$outputDirectory/$jsParserDir"
+    arguments = arguments + listOf("-package", jsParserPkg, "-lib", libDir)
+
+    doFirst {
+        mkdir(libDir)
+    }
+}
+
+tasks.withType<JavaCompile>().configureEach {
+    options.errorprone.excludedPaths.set(".*/(generated-src|$jsParserDir)/.*")
+}
+
+tasks.named("generateGrammarSource") {
+    mustRunAfter(tasks.named("generateEffectiveLombokConfig"))
+}
```

Comment:
I have no idea if this is the best way to do this, it was the first way I found that worked.